modules/controlfreec/1.0/controlfreec.smk

```diff
@@ -0,0 +1,157 @@
+    _controlfreec_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _controlfreec_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+rule _controlfreec_config:
+    input:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam"
+    output:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/config_WGS.txt"
+    conda:
+        CFG["conda_envs"]["sambamba"]
+    params:
+        config = CFG["options"]["configFile"],
+        outdir = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/",
+        window = CFG["options"]["window"],
+        ploidy = CFG["options"]["ploidy"],
+        chrLen = CFG["options"]["chrLenFile"],
+        chrFiles = CFG["options"]["chrFiles"],
+        reference = CFG["options"]["gemMappabilityFile"]
+    shell:
+        "samtoolspath=$(which samtools ) ; "
+        "samtoolsPathName=$(echo $samtoolspath) ; "
+        "sambambapath=$(which sambamba ) ; "
+        "sambambaPathName=$(echo $sambambapath) ; "
+        "bedtoolspath=$(which bedtools ) ; "
+        "bedtoolsPathName=$(echo $bedtoolspath) ; "
+        "sed \"s|BAMFILE|{input.bam}|g\" {params.config} | "
+        "sed \"s|OUTDIR|{params.outdir}|g\" | "
+        "sed \"s|windowSize|{params.window}|g\" | "
+        "sed \"s|ploidyInput|{params.ploidy}|g\" | "
+        "sed \"s|chrLenFileInput|{params.chrLen}|g\" | "
+        "sed \"s|chrFilesPath|{params.chrFiles}|g\" | "
+        "sed \"s|sambambaPath|$sambambaPathName|g\" | "
+        "sed \"s|bedtoolsPath|$bedtoolsPathName|g\" | "
+        "sed \"s|samtoolsPath|$samtoolsPathName|g\" | "
+        "sed \"s|referenceFile|{params.reference}|g\" > {output}"
+
+rule _controlfreec_run:
+    input:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/config_WGS.txt"
+    output:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_info.txt",
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_ratio.txt",
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/{sample_id}.bam_CNVs"
+    conda: CFG["conda_envs"]["controlfreec"]
+    threads: CFG["threads"]["controlfreec_run"]
+    resources: mem_mb = CFG["mem_mb"]["controlfreec_run"]
+    log:
+        stdout = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/run.stdout.log",
+        stderr = CFG["logs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/run.stderr.log"
+    shell:
+        "freec -conf {input} > {log.stdout} 2> {log.stderr} "
+
+rule _controlfreec_calc_sig:
```

Comment:
I don't see how this rule is ever run. It's outputs don't come up in latter rules (or the all rule). What am I missing?

Reply:
Ah you're right. I'll need to add the output into the all rule